src/Https.php

```diff
@@ -133,6 +133,7 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
             $location = parse_url($response->getHeaderLine('Location'));
 
             if (empty($location['host']) || $location['host'] === $uri->getHost()) {
+                $location['host'] = $uri->getHost();
```

Comment:
I think the problem is in the `if` that should be
```php
if (!empty($location['host']) && $location['host'] === $uri->getHost()) {
```

Reply:
Yeah, that would solve it too (I didn't look into the details enough).

Does it actually make sense to redirect traffic that uses a non-standard port to the standard port using https? In normal use cases, a server running on a non-default port, will probably also not have https on the default port.

Depending on whether this is something that you want to keep in, I can either do the requested change, or I could remove it?

Reply:
I don't think there're many use cases in which a https request use a different port than 443. We could include an option to use a different port, but while nobody ask for this feature, I'd keep it as is.